src/app/components/multiple-draft-generator/multiple-draft-generator.component.ts

```diff
@@ -0,0 +1,60 @@
+import {Component} from '@angular/core';
+import {Resource} from '../../models/resource';
+import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
+import {DraftService} from '../../service/draft.service';
+import {Translation} from '../../models/translation';
+
+@Component({
+  selector: 'admin-multiple-draft-generator',
+  templateUrl: './multiple-draft-generator.component.html'
+})
+export class MultipleDraftGeneratorComponent {
+  resource: Resource;
+  translations: Translation[];
+
+  confirmMessage: string;
+  saving: boolean;
+  errorMessage: string;
+
+  constructor(private ngbActiveModal: NgbActiveModal, private draftService: DraftService) {}
+
+  showConfirmAlert(): void {
+    this.translations = this.resource.translations.filter(translation => translation.generateDraft);
+    if (this.translations.length === 0) {
+      return;
+    }
+    this.confirmMessage = 'Are you sure you want to generate a draft for these languages: ';
+
+    this.translations.forEach((translation, index) => {
```

Comment:
I think it could be cleaner to use [join](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join) on the array and then just append to the string once. What you have works though.

Here's an example:
```js
const translationsString = this.translations
    .map(translation => translation.language.name)
    .join(', ');

this.confirmMessage = `Are you sure you want to generate a draft for these languages: ${translationsString}?`;
```